sdk/python/symbolchain/BasicTransactionFactory.py

```diff
@@ -44,3 +44,23 @@ def _create(self, transaction_descriptor, factory_class):
 		processor.copy_to(transaction, ['type', 'signer_public_key'])
 
 		return transaction
+
+	@staticmethod
+	def _encode_string_field(obj, key):
+		value = getattr(obj, key)
+		if isinstance(value, str):
+			setattr(obj, key, value.encode('utf8'))
+
+	def _auto_encode_strings(self, transaction_descriptor, transaction):
+		all_type_hints = build_type_hints_map(transaction)
+
+		# auto encode "top-level" strings
+		# _todo: maybe instead of doing this, there should be hint on: `namespaceRegistration.name` and `transfer.message`
+		for key in transaction_descriptor.keys():
+			if 'type' == key:
+				key = 'type_'
```

Comment:
why this check ?

Reply:
difference between transaction and descriptor

(and type hints contains hint for "type_" not "type")